change/@ni-nimble-angular-fb64a60a-b575-4965-bea7-df33b3911b9f.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
@kjohn1922, I'm moving this question to make it easier to have a discussion:

> "Performance Audit with Lighthouse" is failing the build. I don't see anything about Lighthouse in the README or CONTRIBUTING docs. Is there something I need to do that I'm missing?

I haven't seen that before, and when I run Lighthouse against the Angular app, I see a significantly higher performance score. I don't have any great suggestions other than re-running the pipeline to see if it was just some fluke. Perhaps others have a better suggestion -- @jattasNI @rajsite

Reply:
I've seen similar very low frequency intermittent low lighthouse scores before (usually from chromatic server issues), a new commit was pushed and is running so lets see what happens

Reply:
@mollykreis @rajsite It's still failing after running it a couple of times this morning. I'm a bit at a loss on how to troubleshoot the issue.

Reply:
Looks like in the lighthouse score the accessibility score is failing to calculate altogether after the change:
![image](https://github.com/ni/nimble/assets/1588923/e6f489da-2b51-4bfe-a768-913ee40b22e3)

Some ideas:
- Try and reverse engineer what change is causing the issue. You can run make changes locally and re-run light house with `npm run performance` or by starting the Angular dev server and running lighthouse in chrome devtools: https://developer.chrome.com/docs/lighthouse/overview/#devtools
- It looks like an error in lighthouse itself is causing the failure. Maybe updating it could fix the issue? From `npm outdated` you can see the version we are using is a bit out of date: 
![image](https://github.com/ni/nimble/assets/1588923/45004fc2-1ce9-45e2-b75a-ff2f23251124)

Reply:
@jattasNI @kjohn1922 my merge from main did two things, update lighthouse and change the accessibility status check to warn (because the upgrade caused it to go below 88 with new checks).

After merging in this branch it does not look like it fixed the error being introduced in this PR that causes the accessibility score to fail to calculate altogether: https://storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/1699290639438-32861.report.html

So I'm still interested in seeing if you are able to narrow down what causes the score to fail to calculate from the changes. You might try reducing to as simple a nimble-card as possible and incrementally adding child elements until you can identify the issue.

While we have allowed a lowering of the accessibility score to be allowed through, I'm not sure about allowing something through that completely breaks the calculation. Thoughts?

Reply:
Nice! Thanks for pushing the update to my branch.
![image](https://github.com/ni/nimble/assets/35350751/3221d09e-f09f-4a84-b9a6-aed1a4fca405)

It's still getting some questionable results:
![image](https://github.com/ni/nimble/assets/35350751/30674174-0f48-4a19-84a9-e212080d5593)
I see low scores if I run it locally against `main`, but I've only seen the `NaN` result in my branch, for some reason. I get that NaN even if I remove the new card component from the example app altogether, though, so I don't think it's anything I'm doing that's causing the issue.

Reply:
@kjohn1922 yep, I think the NaN result corresponds to my comment saying accessibility fails to calculate (might be another case of simultaneous posting if you haven't seen my comment :P)

Reply:
Yep, I posted a reply without seeing your latest comment. Not sure how that keeps happening, haha. I'll look into the NaN value some more.

Reply:
Still debugging, but I'm not sure what to make of this: If I remove `aria-labelledby="title-slot"` from the `section` in the card template, the issue goes away and I get a value instead of NaN from the accessibility check.

Reply:
Huh, wonder if the library  reporting the error is making bad assumptions about ID formats. Looks like other IDs snuck by formatted as camelCase instead of kebab-case. Could try switching back to that, ie titleSlot

Reply:
Tried camel case, that was not the problem.

It gets a value, though, if I explicitly set `role="region"` on the `section`. The Lighthouse report was actually trying tell me what was wrong, I think:
![image](https://github.com/ni/nimble/assets/35350751/870fed3e-c722-4a13-a340-88ae30156fd6)

A `section` implicitly has a `role` of `region`, but I think it's looking for it to be set explicitly. I feel like I'm trying to read between the lines, though, and I'm not sure why that results in a NaN value rather than a hit to the score. The other issues listed in the report don't have the "Error!" text, though, and that error goes away if I add `role="region"`.

Reply:
>  I feel like I'm trying to read between the lines

It's possible the axe extension would give more info, built on the same library: https://www.deque.com/axe/browser-extensions/

> The other issues listed in the report don't have the "Error!" text, though, and that error goes away if I add role="region".

That sounds like an okay workaround to unblock the PR! Can leave a comment and reference to the issue in the template.

Reply:
Pushed a change with the role set and a code comment pointing to the issue.